src/librustc_mir/shim.rs

```diff
@@ -98,14 +99,25 @@ fn make_shim<'a, 'tcx>(tcx: ty::TyCtxt<'a, 'tcx, 'tcx>,
         ty::InstanceDef::DropGlue(def_id, ty) => {
             build_drop_shim(tcx, def_id, ty)
         }
+        ty::InstanceDef::CloneShim(def_id, ty) => {
+            let name = tcx.item_name(def_id).as_str();
+            if name == "clone" {
+                build_clone_shim(tcx, def_id, ty)
+            } else if name == "clone_from" {
+                debug!("make_shim({:?}: using default trait implementation", instance);
+                return tcx.optimized_mir(def_id);
```

Comment:
Actually, it does not do anything that's useful for our case - we already don't need to write bounds-checks because we''re cloning arrays.